### ISSUE TITLE
virtual cigarette

### ISSUE THREAD

**Issue by maddash1337** [maelstrom2001/ArcticFox#176]
I would like to have a virtual cigarette feature.
I am thinking about a option to limit the puffs for lets say a 10 minutes timeframe to 20 puffs. After 10 minutes the counter and timer should rest until the next puff. Also it would be nice if you can decide what timeframe and puff limit .

**Comment by docdeath** [comment]
really  virtual cig  ? not being funny why vaping is to help give up smoking so why would you want to over complected vape mod with some thing that is  not needed   20 puffs in  10mins lol

**Comment by maddash1337** [comment]
Because sometimes I join some smokers at work. And if I use my vape while they smoking I will vape more than I need. So it will reduce my amount of puffs.... At least I think so.

**Comment by Dil3mm4** [comment]
I feel too @maddash1337 request, would be a nice feature to implement.

**Comment by CCCheshire** [comment]
Totally agree! I desperately need this option too. Or little simplier - just possibility to set number of puffs to lock device. After locking puffs must be automatically set to 0 and after unlocking new vaping session can be started.

**Comment by maddash1337** [comment]
Yeah would also be fine for me. Simpler - no timer required ... sounds good.

**Comment by GomezAddams** [comment]
I agree with the **CCCheshire**'s formulation. It would be very nice if such option will be implemented.

**Comment by Dil3mm4** [comment]
@TBXin @maelstrom2001 isn't this a questionable feature? :) 
Setting X number of puffs then lock (or a simple message) the device!

**Comment by ilffej** [comment]
I think this is a good idea - maybe have an option to set auto shut down after a certain number/duration of puffs.

**Comment by sala91** [comment]
I love this idea. Atm, vaping is like lesbian sex - you never know when to end...
